src/connection.ts

```diff
@@ -2484,7 +2485,7 @@ class Connection extends EventEmitter {
     }
 
     payload.hostname = this.config.options.workstationId || os.hostname();
-    payload.serverName = this.routingData ? this.routingData.server : this.config.server;
+    payload.serverName = this.routingData ? `${this.routingData.server}\\${this.routingData.instance}` : this.config.server;
```

Comment:
Is `instance` here guaranteed to always be a non-empty string? What happens if it's empty?